### DISCUSSION TITLE
Architecture Design: Where to put common functions, service calls?

### DISCUSSION THREAD

**Discussion by a-h-abid** [nWidart/laravel-modules#1410] [Q&A] [answered]
Hi,

I have a small query regarding structuring modules & common services.

I have some modules where they have some common functions (like helpers, utilities, external services etc.) shared. I currently kept them at App namespace as it seems to be a good place to me.

What would be the ideal place to put these common functions in?
- In App namespace as I'm currently doing?
- In a common Module?
- Something else?

**Comment by dcblogdev** [comment, marked as answer]
I think they are fine to go into app, I'll often do that or create an admin module that acts as a global place for all my application specific code to go. Either if fine I feel like app seems like a better place
